framework/src/application.ts

```diff
@@ -229,6 +229,11 @@ export class Application {
 		plugin: BasePlugin<any>,
 		options: PluginConfig = { loadAsChildProcess: false },
 	): void {
+		for (const registeredModule of this._registeredModules) {
```

Comment:
I think similar check needs to exist on `registerModule` as well?

Reply:
I am not sure. Is the order of registering modules & plugins not fixed? 
If it is fixed i.e. all modules are always registered before registering a plugin, it may not be necessary.

Reply:
the order is not fixed, the function can be called in any order